optuna/samplers/_grid.py

```diff
@@ -106,10 +104,8 @@ def __init__(self, search_space: Mapping[str, Sequence[GridValueType]]) -> None:
                 self._check_value(param_name, value)
 
         self._search_space = collections.OrderedDict()
-        for param_name, param_values in sorted(search_space.items(), key=lambda x: x[0]):
-            param_values = cast(SortableParamValueSequenceType, param_values)
-
-            self._search_space[param_name] = sorted(param_values)
+        for param_name, param_values in sorted(search_space.items()):
```

Comment:
```suggestion
        for param_name, param_values in sorted(search_space.items(), key=lambda x: x[0]):
```
Is there some reason to sort by `param_values` also?

Reply:
This does not change the behavior because `param_name`s are distinct.

Reply:
Yes if `x[1]` is of a sortable type, but if `x[1] is None`, wouldn't that be an error?

Reply:
(I am not sure whether the lexicographical comparison in Python is strict or lazy, but just for mental ease :) )

Reply:
`search_space.items()` is a sequence of tuples of `str` and `Sequence[GridValueType]`s, and a `Sequence[GridValueType]` may not be order-comparable (a generator would be an example). While it is true that there would never be overlapping keys and the `Sequence` element never participates in comparison (thus there would never be a `TypeError`), I think it would be more readable if you explicitly specified the sorting key to be the first element, or at least left some comment.

Reply:
(It's just my mental type checker complaining about the code, and if you think it is more readable this way, I'll merge this as it is.)

Reply:
Thank you for your explanation! Hmm, I guess other parts in Optuna do a similar sort without explicit key specification. So the current code looks fine.